shoop/campaigns/models/campaigns.py

```diff
@@ -75,13 +68,12 @@ def is_available(self):
         return True
 
     def save(self, **kwargs):
-        if self.discount_percentage and self.discount_amount_value:
-            raise ValidationError(_("You should only define either discount percentage or amount."))
-
-        if not (self.discount_percentage or self.discount_amount_value):
-            raise ValidationError(_("You must define discount percentage or amount."))
-
         super(Campaign, self).save(**kwargs)
+        if not self.effects:
+            raise ValidationError(_("At least one effect must be defined."))
```

Comment:
They are on clean also, this is the same functionality there was before. This here is to prevent direct saving without effects.